build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/RestTestBasePlugin.java

```diff
@@ -113,7 +113,7 @@ public void apply(Project project) {
         configureArtifactTransforms(project);
 
         // Create configuration for aggregating historical feature metadata
-        Configuration featureMetadataConfig = project.getConfigurations().create(FEATURES_METADATA_CONFIGURATION, c -> {
+        FileCollection featureMetadataConfig = project.getConfigurations().create(FEATURES_METADATA_CONFIGURATION, c -> {
```

Comment:
referenced from SystemPropertyCommandLineArgumentProvider during execution time. Therefore needs to be configuration cache compatible.